homeassistant/components/here_travel_time/sensor.py

```diff
@@ -0,0 +1,285 @@
+        hass.data.setdefault(DATA_KEY, [])
+
+        travel_mode = config.get(CONF_MODE)
+        traffic_mode = config.get(CONF_TRAFFIC_MODE)
+        route_mode = config.get(CONF_ROUTE_MODE)
+
+        name = config.get(CONF_NAME, DEFAULT_NAME)
+        app_id = config.get(CONF_APP_ID)
+        app_code = config.get(CONF_APP_CODE)
+        origin = config.get(CONF_ORIGIN)
+        destination = config.get(CONF_DESTINATION)
+
+        sensor = HERETravelTimeSensor(
+            hass,
+            name,
+            app_id,
+            app_code,
+            origin,
+            destination,
+            travel_mode,
+            traffic_mode,
+            route_mode,
+        )
+        hass.data[DATA_KEY].append(sensor)
+
+        if sensor.valid_api_connection:
+            add_entities_callback([sensor])
+
+    # Wait until start event is sent to load this component.
+    hass.bus.listen_once(EVENT_HOMEASSISTANT_START, run_setup)
+
+
+class HERETravelTimeSensor(Entity):
+    """Representation of a HERE travel time sensor."""
+
+    def __init__(
+            self,
+            hass,
+            name,
+            app_id,
+            app_code,
+            origin,
+            destination,
+            travel_mode,
+            traffic_mode,
+            route_mode,
+    ):
+        """Initialize the sensor."""
+        import herepy
+
+        self._hass = hass
+        self._name = name
+        self._travel_mode = travel_mode
+        self._traffic_mode = traffic_mode
+        self._route_mode = route_mode
+        self._unit_of_measurement = 'min'
+        self._response = None
+        self.valid_api_connection = True
+
+        # Check if location is a trackable entity
+        if origin.split('.', 1)[0] in TRACKABLE_DOMAINS:
+            self._origin_entity_id = origin
+        else:
+            self._origin = origin
+
+        if destination.split('.', 1)[0] in TRACKABLE_DOMAINS:
+            self._destination_entity_id = destination
+        else:
+            self._destination = destination
+
+        self._client = herepy.RoutingApi(app_id, app_code)
+        self.update()
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        if self._response is None:
+            return None
+
+        # pylint: disable=E1101
+        _summary = self._response.response['route'][0]['summary']
+        return round(_summary['trafficTime'] / 60)
+
+    @property
+    def name(self):
+        """Get the name of the sensor."""
+        return self._name
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        if self._response is None:
+            return None
+
+        # pylint: disable=E1101
+        _summary = self._response.response['route'][0]['summary']
+        # pylint: disable=E1101
+        _route = self._response.response['route']
+
+        res = {}
+        res['distance'] = _summary['distance']
+        res['distance_unit'] = LENGTH_METERS
```

Comment:
might need to convert this for different config settings

Reply:
Not quite sure what you mean here. If you mean the distance_unit: Meters is currently the only unit which is returned by the here API. Even if you request the response to be in imperial units. Until this is resolved I prefer to have this hardcoded here.

Reply:
We should be converting to the units that hass is set to so kolimeters and miles

Reply:
Implemented